product_packaging_level/models/product_packaging.py

```diff
@@ -116,15 +146,31 @@ def _make_qty_per_level(self, qty_per_level_mapping, format_pattern=None):
             res.append((code, new_qty))
         return res
 
-    @api.onchange("packaging_level_id")
+    @api.onchange(
+        "package_type_id",
+    )
+    def _onchange_package_type(self):
+        package_type_level_id = self.package_type_id.packaging_level_id
+        if package_type_level_id:
+            self.packaging_level_id = package_type_level_id
+
+    @api.onchange("packaging_level_id", "package_type_id", "name_type", "name")
     def _onchange_name(self):
-        if self.packaging_level_id:
-            self.name = self.packaging_level_id.name
+        new_name = self.name
+        if self.packaging_level_id and self.name_type == "by_package_level":
```

Comment:
```suggestion
        if self.name_type == "by_package_level":
```
It helps to clear the name if there is no packaging_level_id

Reply:
updated